### ISSUE TITLE
Setting labels.color to the same value changes what is rendered

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#2479]
Setting the color on a labels layer to the current value of `Labels.color` changes the rendering completely, the layer becomes invisible on screen. This is happening because `Layer.color_mode` is being changed in the [`Layer.color` setter](https://github.com/napari/napari/blob/master/napari/layers/labels/labels.py#L353-L374)

I would expect that getting the value of a property then setting the same value should leave the rendering unchanged

setup:
```python
blobs = data.binary_blobs(length=128, volume_fraction=0.1, n_dim=3)
viewer = napari.Viewer(ndisplay=3)
labels = ndi.label(blobs)[0]
labels_layer = viewer.add_labels(labels, name='blob ID')
initial_state = labels_layer._get_state()

```

example:
```python
print(labels_layer.color, labels_layer.color_mode)
{0: array([0., 0., 0., 0.], dtype=float32), None: array([0., 0., 0., 1.], dtype=float32)} auto
labels_layer.color = labels_layer.color
print(labels_layer.color, labels_layer.color_mode)
{0: array([0., 0., 0., 0.], dtype=float32), None: array([0., 0., 0., 1.], dtype=float32)} direct
```

before setting `labels_layer.color`:
<img width="200" alt="Screenshot 2021-03-26 at 21 57 34" src="https://user-images.githubusercontent.com/7307488/112697288-25236700-8e7f-11eb-8544-a0f5aced6ddf.png">

after setting `labels_layer.color`:
<img width="200" alt="Screenshot 2021-03-26 at 21 58 28" src="https://user-images.githubusercontent.com/7307488/112697294-2785c100-8e7f-11eb-9912-5e55bc932314.png">

**Comment by sofroniewn** [comment]
That's no good @alisterburt! Can you see an easy way to fix? Would be nice to have to update `color_mode` at all there. Looks like it was done in #1339 and improved in #1362. If so a PR would be welcome :-)

**Comment by jni** [comment]
Closed by #3275
